modules/gdscript/gdscript_compiler.cpp

```diff
@@ -165,7 +165,30 @@ GDScriptDataType GDScriptCompiler::_gdtype_from_datatype(const GDScriptParser::D
 						}
 					}
 					if (result.script_type_ref.is_null()) {
-						result.script_type_ref = GDScriptCache::get_shallow_script(p_datatype.script_path, main_script->path);
+						if (classes.size() > 1) {
+							auto find_subclass_script = [&classes](Ref<GDScript> lp_script) -> Ref<GDScript> {
+								for (int i = 1; i < classes.size(); ++i) {
+									if (lp_script.is_null()) {
+										return Ref<GDScript>();
+									}
+									HashMap<StringName, Ref<GDScript>>::ConstIterator E = lp_script->get_subclasses().find(classes[i]);
+									if (!E) {
+										return Ref<GDScript>();
+									}
+									Ref<GDScript> tmp_script = lp_script; // because inner_script may hold the only ref to is subclass...
```

Comment:
```suggestion
									Ref<GDScript> tmp_script = lp_script; // Because inner_script may hold the only ref to its subclass...
```
Same typo in `gdscript.cpp`.

Reply:
👍